examples/TUTORIAL.md

```diff
@@ -0,0 +1,253 @@
+
+# Copy the S2I scripts to /usr/libexec/s2i, since openshift/base-centos7 image sets io.openshift.s2i.scripts-url label that way, or you could update the label and copy the s2i scripts in your chosen path.
+COPY ./.s2i/bin/ /usr/libexec/s2i
+
+# Copy our nginx configuration file that omits the user directive. We do this so that we can run the process as a non-root user.
+COPY ./etc/ /opt/app-root/etc
+
+# Change user priviliges away from root and from nginx user and set them to our default user.
+# We will also create a run directory to place the nginx process pid there so that we avoid
+# changing /var/run/ user permissions.
+RUN chown -R 1001:1001 /var/log/nginx/ && chown -R 1001:1001 /var/lib/nginx/ \
+  && mkdir -p /opt/app-root/run && chown -R 1001:1001 /opt/app-root/run
+
+# This default user is created in the openshift/base-centos7 image
+# For security reasons we use it to run the image so that we avoid running as root user.
+USER 1001
+
+# Set the default port for applications built using this image
+EXPOSE 8080
+
+# Set the default CMD for the image
+CMD ["usage"]
+```
+After we are done with that, we will modify a little the Nginx configuration file to make the process run as unprivileged user. We will place this file in our project root in `etc/`.
+```nginx
+worker_processes auto;
+error_log /var/log/nginx/error.log;  # we already changed user privileges for this folder in our docker file, so we can leave the logs to the default location.
+pid /opt/app-root/run/nginx.pid; # change the default place where the pid is placed so that we don't have to modify the whole /var/run directory
+
+events {
+    worker_connections 1024;
+}
+
+http {
+    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                      '$status $body_bytes_sent "$http_referer" '
+                      '"$http_user_agent" "$http_x_forwarded_for"';
+
+    access_log  /var/log/nginx/access.log  main;
+
+    sendfile            on;
+    tcp_nopush          on;
+    tcp_nodelay         on;
+    keepalive_timeout   65;
+    types_hash_max_size 2048;
+
+    include             /etc/nginx/mime.types;
+    default_type        application/octet-stream;
+
+    # Load modular configuration files from the /etc/nginx/conf.d directory.
+    # See http://nginx.org/en/docs/ngx_core_module.html#include
+    # for more information.
+    include /opt/app-root/etc/conf.d/*.conf;
+
+    server {
+        listen       8080 default_server;
+        server_name  _;
+        root         /opt/app-root/src; # tell nginx to load the main html application files from our /opt/app-root/src folder.
+
+        # Load configuration files for the default server block.
+        include /opt/app-root/etc/default.d/*.conf;
+
+        location / {
+        }
+
+        error_page 404 /404.html;
+            location = /40x.html {
+        }
+
+        error_page 500 502 503 504 /50x.html;
+            location = /50x.html {
+        }
+    }
+}
```

Comment:
That's very lengthy. I'd rather seeing short listings, easily graspable, have you seen my article in the topic https://blog.openshift.com/create-s2i-builder-image/ ? Why not using lighttpd here instead of nginx, which is simpler to setup?